doc: addition of a code of conduct

Thea-Aldrich opened the pull request:
This change adds a Code of Conduct based on
https://www.contributor-covenant.org/. Additional reporting and
enforcement guidelines will be added at a later date.

Signed-off-by: Thea Aldrich <aldrich.thea@gmail.com>

Thea-Aldrich committed:
doc: addition of a code of conduct

This change adds a Code of Conduct based on
https://www.contributor-covenant.org/. Additional reporting and
enforcement guidelines will be added at a later date.

Signed-off-by: Thea Aldrich <aldrich.thea@gmail.com>

dbkinder commented on CODE_OF_CONDUCT.rst:
```diff
@@ -0,0 +1,76 @@
+
+We strive to maintain a respectful and professional environment in the Zephyr
+Community. If you experience or witness Code of Conduct violations and wish to
+report the incident please contact us at info@zephyrproject.org.  
```
The project "sanitycheck" has some rules about file changes that enforce good habits. One was caught here: trailing white space is not allowed, there are a couple of spaces at the end of this line you should delete, reported by https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/23650/5/tests

nashif commented:
> This change adds a Code of Conduct based on contributor-covenant.org. Additional reporting and enforcement guidelines will be added at a later date.

I would add that as is, the issue with enforcement was related to the original proposal wrt to the 3 strike enforcement. This one here is acceptable I guess.

nashif commented:
@Thea-Aldrich ping

Thea-Aldrich committed:
doc: Fixing trailing whitespace.

Remove trailing whitespace from new code of conduct section.

Signed-off-by: Thea Aldrich <aldrich.thea@gmail.com>

nashif commented:
> Additional reporting and
> enforcement guidelines will be added at a later date.

@Thea-Aldrich can you please add https://www.contributor-covenant.org/version/1/4/code-of-conduct.md as is, I think we should adopt it without changes.

Thea-Aldrich committed:
doc: add enforcement section

Add enforcement section as requested by tsc.

Signed-off-by: Thea Aldrich <aldrich.thea@gmail.com>

nashif commented on CODE_OF_CONDUCT.rst:
```diff
@@ -0,0 +1,86 @@
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at [INSERT EMAIL ADDRESS]. All
```
Can you add 2 names here, mine and Kate's as agreed on in the F2F meeting

nashif commented:
Can this be updated please?

Thea-Aldrich committed:
doc: add reporting address to CoC

This change adds an email alias for Code of Conduct reports and
identifies the recipients of reports.

Signed-off-by: Thea Aldrich <aldrich.thea@gmail.com>
